### ISSUE TITLE
Consider allowing header[role=banner]

### ISSUE THREAD

**Issue by WilcoFiers** [dequelabs/axe-core#3108]
```html
  <main>
    <header role="banner">
      Hello world
    </header>
  </main>
```

This fails `aria-allowed-role`, in addition to failing `landmark-banner-is-top-level`. It's probably better for axe-core to only fail it under the top-level rule. It's a little odd to allow role=banner on a `header` if it's at the top, but to fail it if it's inside another landmark.

The same problem exists for other landmarks. I think the fix is to add `banner` to the header element in standards/html-elms.js.

**Comment by WilcoFiers** [comment]
Actually it'd be pretty cool if axe-core explicitly reported on the top-level rule to remove the role attribute.

**Comment by padmavemulapati** [comment]
Validated with the latest develop branch code base, now its showing only one rule failing for the code snippet
```
<main>
    <header role="banner">
      Hello world
    </header>
  </main>
```
when header is having `role=banner` 
![image](https://user-images.githubusercontent.com/47409579/187240247-10ec140e-3a2a-4a8e-867b-2cac795e9ab7.png)

Previouly both `aria-allowed-role` and `landmark` rules failing.
<img width="1461" alt="image" src="https://user-images.githubusercontent.com/47409579/187240439-9cec7219-0c2b-4259-9aa6-6dd2cc15eacc.png">
